packages/reactDom/src/ReactDOM.ml

```diff
@@ -249,7 +250,19 @@ let add kind value map =
 type dangerouslySetInnerHTML = < __html : string >
 
 (* `Booleanish_string` are JSX attributes represented as boolean values but rendered as strings on HTML https://github.com/facebook/react/blob/a17467e7e2cd8947c595d1834889b5d184459f12/packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js#L1165-L1176 *)
-let booleanish_string name v = React.JSX.string name (string_of_bool v)
+let getPropByName name =
+  match DomProps.findByName name with
+  | Ok prop -> (name, DomProps.getJSXName prop)
+  | Error _ -> failwith "Invalid prop"
+
+let string name v = React.JSX.string (getPropByName name) v
+let int name v = React.JSX.int (getPropByName name) v
+let bool name v = React.JSX.bool (getPropByName name) v
+
+let booleanish_string name v =
+  React.JSX.string (getPropByName name) (string_of_bool v)
+
+let float name v = React.JSX.float (getPropByName name) v
```

Comment:
We really don't want this, getPropByName is more expensive than the direct name, we always prefer to keep it static here